### PR TITLE
WIP: WRED ECN support

### PR DESCRIPTION
**Pull request by spenugondaa**
WIP: This pull request is for CLI template to support wred profile management.
WRED profile takes the following parameters: profile-name, red-min-threshold, red-max-threshold, red-drop-rate, green-min-threshold, green-max-threshold, green-drop-rate, yellow-min-threshold, yellow-max-threshold, yellow-drop-rate and ecn.
As there is no oc yang available, decision has been made to use the sonic yang.

`sonic-wred-profile.yang` contains the model for the profiles.
`src/CLI/clitree/cli-xml/wred.xml` contains the CLI template
`src/CLI/actioner/sonic-cli-wred.py` contains the actioner script for get/set/delete
`src/CLI/renderer/templates/show_wred.j2` contains renderer script.

example:
```
sonic# show wred
---------------------------------------------------
Profile                : myprof4
---------------------------------------------------
Profile                : new3
green-max-threshold    : 100
---------------------------------------------------
Profile                : newprof1
yellow-max-threshold   : 10

sonic(config)# wred prof1
sonic(conf-wred-prof1)#
  ecn                               Configure ecn of the profile
  green-drop-rate          Configure green drop rate of the profile
  green-max-threshold   Configure green maximum threshold of the profile
  green-min-threshold   Configure green minimum threshold of the profile
  no                                 Negate a command or set its defaults
  red-drop-rate               Configure red drop rate of the profile
  red-max-threshold       Configure red maximum threshold of the profile
  red-min-threshold       Configure red minimum threshold of the profile
  yellow-drop-rate         Configure yellow drop rate of the profile
  yellow-max-threshold  Configure yellow maximum threshold of the profile
  yellow-min-threshold  Configure yellow minimum threshold of the profile
sonic(conf-wred-prof1)# green-max-threshold 10
sonic(conf-wred-prof1)# exit
sonic# show wred
---------------------------------------------------
Profile                : myprof4
---------------------------------------------------
Profile                : new3
green-max-threshold    : 100
---------------------------------------------------
Profile                : newprof1
yellow-max-threshold   : 10
---------------------------------------------------
Profile                : prof1
green-max-threshold    : 10
```

### CODE REVIEW

**Comment by jeffyin-dell** [comment]
@spenugondaa -- do you need to update your SONiC YANG files to match the _new_ SONiC YANG format?

**Comment by spenugondaa** [comment]
> @spenugondaa -- do you need to update your SONiC YANG files to match the _new_ SONiC YANG format?

@jeff-yin not for new sonic format but we need to modify the yang file a bit based on the internal HLD review comment. @eddyatdell  will work on it and push the updated yang tomorrow.

File: models/yang/sonic/sonic-wred-profile.yang [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC WRED_PROFILE";
+
+    revision 2019-05-15 {
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-wred-profile.yang:14]: Update the date.

File: models/yang/sonic/sonic-wred-profile.yang [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        container WRED_PROFILE {
+            list WRED_PROFILE_LIST {
+                key "name";
+
+                leaf name{
+                    type string;
+                }
+
+                leaf yellow_min_threshold {
+                    type uint64;
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-wred-profile.yang:29]: Define constraint like range of values or special value etc. Take care in all places applicable.

**Review by dutta-partha — CHANGES_REQUESTED** [review]

File: models/yang/sonic/sonic-wred-profile.yang [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            "Initial revision.";
+    }
+
+    container sonic-wred-profile {
+        container WRED_PROFILE {
+            list WRED_PROFILE_LIST {
+                key "name";
+
+                leaf name{
+                    type string;
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-wred-profile.yang:25]: Any specific pattern of the name ?

**Comment by bhavini-gada** [comment]
Closing as this feature is not planned for Buzznik anymore.

**Pull request closed without merging**